Tail  whip logic fix and logging

MarsM0nd opened the pull request:
Were only able to tail whip while being both restrained and buckled, when the intention probably was neither restrained nor buckled. Also adds logging for the tail whip attack.
:cl:
fix: Unathi can now tailwhip when they are supposed to be able to.
add: Tail whipping gets logged.
/:cl:

davidchan13 commented:
The original intent was that it /could/ be used while restrained, but buckled might be a tad to far.

Citinited commented:
I dunno, the original code does state, `You need freedom of movement to tail lash!`

MarsM0nd commented:
Yeah, i would figure that if it can be used while restrained, it wouldn't just have been checked for it.

KasparoVy commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -93,7 +93,7 @@
 /datum/action/innate/tail_lash/Activate()
 	var/mob/living/carbon/human/user = owner
-	if(!user.restrained() || !user.buckled)
+	if(user.restrained() || user.buckled)
```
If you want a restrained Unathi that isn't being pulled by someone (i.e. has cuffs on but can still move) to be able to do this, you just have to amend this line so it's 

`if((user.pulledby && user.restrained()) || user.buckled)`

tigercat2000 replied:
Do this please ^

Aurorablade commented:
Huh it was meant to so that if you are buckled and restrained you can't tail whip. I am iffy on letting them do it while cuffed as I would imagine they need arms for balancing them selves while spining to whip

KasparoVy commented:
Then include a chance to fall with `prob(50)` and `weaken(5)` if you're cuffed.

Aurorablade commented:
seems alright of an idea,

MarsM0nd commented:
Added the chance to weaken yourself if lashing while cuffed, and added the flags that makes it unusable while stunned, lying on the floor or dead. As funny as it might be having an Unathi corpse suddenly spin and hit the coroner with their tail, that's probably not wanted.

davidchan13 commented:
>As funny as it might be having an Unathi corpse suddenly spin and hit the coroner with their tail, that's probably not wanted.

Elon Musk cancels all plans to visit Mars. Declares it a barren, dead planet of no fun. /s

KasparoVy commented:
Fantastic

taukausanake commented:
> Unathi corpse suddenly spin and hit the coroner with their tail

Reminds me of lizards with removable tails, that's a feature Unathi should have. Using their tail as a whipping weapon

KasparoVy commented:
Lmfao. If tails get refactored to be organs, maybe.

Citinited commented:
Actually, this would be perfect if you could chop off vulp / unathi / tajaran tails to prevent this. 🤔

Fox-McCloud commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -90,10 +90,11 @@
 /datum/action/innate/tail_lash/Activate()
 	var/mob/living/carbon/human/user = owner
-	if(!user.restrained() || !user.buckled)
+	if((user.restrained() && user.pulledby) || user.buckled)
```
why not just check for `user.incapacitated()`?

MarsM0nd replied:
The check for other incapacitations shouldn't be needed, as the check_flags should handle everything else.

alexkar598 commented:
@MarsM0nd is this stale?

MarsM0nd commented:
Added the return to stop the tail lashing when you loose your balance. Also added a check and return if enough people are hit to take a good hit on your stamina.

KasparoVy commented:
:thinking: Looks done

alexkar598 commented:
CL is not done

Needs:

Tweak:buffs Slith

KasparoVy commented:
I have a dream where the tailwhip icon isn't just there to make us feel better

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -108,7 +109,15 @@
+					user.visible_message("<span class='danger'>[src] lost their balance! </span>", "<span class='danger'>You lost your balance!</span>")
+					return
+			if(user.getStaminaLoss() >= 60) //Bit higher as you don't need to start, just would need to keep going with the tail lash.
+				to_chat(user, "<span class='warning'>You did hit quite a few people with your tail, and lost the energy to continue.</span>")
```
Change this to `<span class='warning'>You run out of momentum!</span>`

Fox-McCloud commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -108,7 +109,15 @@
+					return
+			if(user.getStaminaLoss() >= 60) //Bit higher as you don't need to start, just would need to keep going with the tail lash.
+				to_chat(user, "<span class='warning'>You did hit quite a few people with your tail, and lost the energy to continue.</span>")
+				return
```
Trailing return

MarsM0nd replied:
Not sure what you mean, that return there is needed to break out of the for loop.

tigercat2000 replied:
Mars is correct.

MarsM0nd commented:
Text changed, definitly better than what i had put there. Short and to the point.

tigercat2000 commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -108,7 +109,15 @@
+			if(user.restrained())
+				if(prob(50))
+					user.Weaken(5)
+					user.visible_message("<span class='danger'>[src] lost their balance! </span>", "<span class='danger'>You lost your balance!</span>")
```
Change this to `"<span class='danger'>[src] loses [p_their()] balance!</span>", "<span class='danger'>You lose your balance!</span>"

KasparoVy replied:
New pronouns ftw

Fox-McCloud commented:
please update

Fox-McCloud commented on code/modules/mob/living/carbon/human/species/station.dm:
```diff
@@ -108,7 +109,15 @@
+			if(user.restrained())
+				if(prob(50))
+					user.Weaken(5)
+					user.visible_message("<span class='danger'>[src] loses [p_their()] balance!</span>", "<span class='danger'>You lose your balance!</span>")
```
Tali Lash loses its balance.

KasparoVy commented on code/modules/mob/living/carbon/human/species/unathi.dm:
```diff
@@ -85,8 +86,17 @@
+			if(user.restrained())
+				if(prob(50))
+					user.Weaken(5)
+					user.visible_message("<span class='danger'>[src] loses [p_their()] balance!</span>", "<span class='danger'>You lose your balance!</span>")
```
`[user] loses...`

Fox-McCloud replied:
also action button datums don't have a gender, either.

Fox-McCloud commented:
please fix conflicts and address whats' @KasparoVy and I pointed out.
